scip.proto

```diff
@@ -369,3 +375,88 @@ enum DiagnosticTag {
   Unnecessary = 1;
   Deprecated = 2;
 }
+
+// Language standardises names of common programming languages that can be used
+// for the `Document.language` field. The primary purpose of this enum is to
+// prevent a situation where we have a single programming language ends up with
+// multiple string representations. For example, the C++ language uses the name
+// "C_PlusPlus" in this enum and other names such as "cpp" are incompatible.
+// Feel free to send a pull-request to add missing programming languages.
+enum Language {
+  UnspecifiedLanguage = 0;
+  Java = 1;
+  Scala = 2;
+  Kotlin = 3;
+  Go = 4;
+  TypeScript = 5;
+  JavaScript = 6;
+  Python = 7;
+  Ruby = 8;
+  C_Sharp = 9;
+  C_PlusPlus = 10;
+  Objective_C = 11;
```

Comment:
Any specific reason for mixing underscores instead of sticking to CamelCase? It seems a little odd; I don't think many languages follow this casing convention.